File: tests/apitest.py
import requests
import json

def test_prediction(data, test_name=""):
    """Make a prediction request and print the results"""
    # Try Next.js server first, then Python server as fallback
    urls = [
        "http://localhost:3000/api/predict",
        "http://localhost:8000/predict"
    ]
    
    for url in urls:
        try:
            response = requests.post(
                url,
                headers={"Content-Type": "application/json"},
                json=data
            )
            print(f"\n=== {test_name} ===")
            print(f"URL: {url}")
            print(f"Request: {json.dumps(data, indent=2)}")
            print(f"Response Status: {response.status_code}")
            print(f"Response: {json.dumps(response.json(), indent=2)}")
            return
        except requests.RequestException as e:
            continue
    print(f"\nError: Could not connect to any server for {test_name}")

# Test cases
high_risk_case = {
    "pregnancies": 6,
    "glucose": 195,
    "bloodPressure": 130,
    "skinThickness": 40,
    "insulin": 180,
    "bmi": 37.2,
    "diabetesPedigreeFunction": 0.6,
    "age": 45
}

low_risk_case = {
    "pregnancies": 1,
    "glucose": 85,
    "bloodPressure": 66,
    "skinThickness": 29,
    "insulin": 100,
    "bmi": 23.5,
    "diabetesPedigreeFunction": 0.351,
    "age": 31
}

minimum_values_case = {
    "pregnancies": 0,
    "glucose": 0,
    "bloodPressure": 0,
    "skinThickness": 0,
    "insulin": 0,
    "bmi": 0,
    "diabetesPedigreeFunction": 0,
    "age": 0
}

custom_values_case = {
    "pregnancies": 5,
    "glucose": 117,
    "bloodPressure": 92,
    "skinThickness": 0,
    "insulin": 0,
    "bmi": 34.1,
    "diabetesPedigreeFunction": 0.337,
    "age": 38
}

dataset_values_case = {
    "pregnancies": 10,
    "glucose": 125,
    "bloodPressure": 70,
    "skinThickness": 26,
    "insulin": 115,
    "bmi": 31.1,
    "diabetesPedigreeFunction": 0.205,
    "age": 41
}

negative_case = {
    "pregnancies": 3,
    "glucose": 126,
    "bloodPressure": 88,
    "skinThickness": 41,
    "insulin": 235,
    "bmi": 39.3,
    "diabetesPedigreeFunction": 0.704,
    "age": 27
}

positive_case = {
    "pregnancies": 7,
    "glucose": 196,
    "bloodPressure": 90,
    "skinThickness": 0,
    "insulin": 0,
    "bmi": 39.8,
    "diabetesPedigreeFunction": 0.451,
    "age": 41
}

if __name__ == "__main__":
    test_prediction(high_risk_case, "High Risk Case")
    test_prediction(low_risk_case, "Low Risk Case")
    test_prediction(minimum_values_case, "Minimum Values Case")
    test_prediction(custom_values_case, "Custom Values Case")
    test_prediction(dataset_values_case, "Dataset Values Case")
    test_prediction(negative_case, "Negative Case")
    test_prediction(positive_case, "Positive Case")

File: tests/apitest.cs
using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

public class DiabetesTestData
{
    public int Pregnancies { get; set; }
    public double Glucose { get; set; }
    public double BloodPressure { get; set; }
    public double SkinThickness { get; set; }
    public double Insulin { get; set; }
    public double BMI { get; set; }
    public double DiabetesPedigreeFunction { get; set; }
    public int Age { get; set; }
}

public class ApiTest
{
    private static readonly HttpClient client = new HttpClient();
    private static readonly string[] urls = new[]
    {
        "http://localhost:3000/api/predict",
        "http://localhost:8000/predict"
    };

    public static async Task TestPrediction(DiabetesTestData data, string testName)
    {
        foreach (var url in urls)
        {
            try
            {
                var content = new StringContent(
                    JsonSerializer.Serialize(data),
                    Encoding.UTF8,
                    "application/json"
                );

                var response = await client.PostAsync(url, content);
                var responseBody = await response.Content.ReadAsStringAsync();

                Console.WriteLine($"\n=== {testName} ===");
                Console.WriteLine($"URL: {url}");
                Console.WriteLine($"Request: {JsonSerializer.Serialize(data, new JsonSerializerOptions { WriteIndented = true })}");
                Console.WriteLine($"Response Status: {(int)response.StatusCode}");
                Console.WriteLine($"Response: {responseBody}");
                return;
            }
            catch (Exception ex)
            {
                continue;
            }
        }
        Console.WriteLine($"\nError: Could not connect to any server for {testName}");
    }

    public static async Task Main()
    {
        var highRiskCase = new DiabetesTestData
        {
            Pregnancies = 6,
            Glucose = 195,
            BloodPressure = 130,
            SkinThickness = 40,
            Insulin = 180,
            BMI = 37.2,
            DiabetesPedigreeFunction = 0.6,
            Age = 45
        };

        var lowRiskCase = new DiabetesTestData
        {
            Pregnancies = 1,
            Glucose = 85,
            BloodPressure = 66,
            SkinThickness = 29,
            Insulin = 100,
            BMI = 23.5,
            DiabetesPedigreeFunction = 0.351,
            Age = 31
        };

        var minimumValuesCase = new DiabetesTestData
        {
            Pregnancies = 0,
            Glucose = 0,
            BloodPressure = 0,
            SkinThickness = 0,
            Insulin = 0,
            BMI = 0,
            DiabetesPedigreeFunction = 0,
            Age = 0
        };

        var customValuesCase = new DiabetesTestData
        {
            Pregnancies = 5,
            Glucose = 117,
            BloodPressure = 92,
            SkinThickness = 0,
            Insulin = 0,
            BMI = 34.1,
            DiabetesPedigreeFunction = 0.337,
            Age = 38
        };

        var datasetValuesCase = new DiabetesTestData
        {
            Pregnancies = 10,
            Glucose = 125,
            BloodPressure = 70,
            SkinThickness = 26,
            Insulin = 115,
            BMI = 31.1,
            DiabetesPedigreeFunction = 0.205,
            Age = 41
        };

        var negativeCase = new DiabetesTestData
        {
            Pregnancies = 3,
            Glucose = 126,
            BloodPressure = 88,
            SkinThickness = 41,
            Insulin = 235,
            BMI = 39.3,
            DiabetesPedigreeFunction = 0.704,
            Age = 27
        };

        var positiveCase = new DiabetesTestData
        {
            Pregnancies = 7,
            Glucose = 196,
            BloodPressure = 90,
            SkinThickness = 0,
            Insulin = 0,
            BMI = 39.8,
            DiabetesPedigreeFunction = 0.451,
            Age = 41
        };

        await TestPrediction(highRiskCase, "High Risk Case");
        await TestPrediction(lowRiskCase, "Low Risk Case");
        await TestPrediction(minimumValuesCase, "Minimum Values Case");
        await TestPrediction(customValuesCase, "Custom Values Case");
        await TestPrediction(datasetValuesCase, "Dataset Values Case");
        await TestPrediction(negativeCase, "Negative Case");
        await TestPrediction(positiveCase, "Positive Case");
    }
}